learningequality/studio
# Split `deferred_user_data` into two API endpoints.

## Summary
### Description of the change(s) you made
This pull request addresses the performance issue by splitting the deferred user settings function into two separate functions. `settingsDeferredUserApiToken()` function retrieves the API token, and `settingsDeferredUserSpaceByKind()` retrieves space used by kind with throttling to avoid excessive requests.

The `fetchDeferredUserData()` function has been updated to retrieve both API token and space used by kind separately and then commit the result to the session using `Promise.all`.



### Manual verification steps performed
1. Run the local development server
2. log in as a `user`
3. Go to the `settings` page and notice 2 API endpoints getting called.

### Screenshots (if applicable)
#### Before:
![Screenshot from 2023-03-10 12-52-35](https://user-images.githubusercontent.com/53405133/224250249-d2e7cc40-079d-492f-9156-82cae1203d6d.png)
![Screenshot from 2023-03-10 12-52-42](https://user-images.githubusercontent.com/53405133/224250252-bc72efec-6060-44a0-87ea-647e8aba8ee8.png)

 
#### After:
![Screenshot from 2023-03-10 12-51-53](https://user-images.githubusercontent.com/53405133/224250235-aaeae04c-bef6-44af-b5c0-5e303631c29b.png)
![Screenshot from 2023-03-10 12-52-02](https://user-images.githubusercontent.com/53405133/224250245-2e0249c1-5249-470b-a301-544234b7f6fb.png)
![Screenshot from 2023-03-10 12-52-05](https://user-images.githubusercontent.com/53405133/224250246-e30dc276-8a65-473d-b601-17c6570ea17f.png)


### Does this introduce any tech-debt items?

___
## Reviewer guidance
### How can a reviewer test these changes?



### Are there any risky areas that deserve extra testing?



## References

#3965 

## Comments


----

### Contributor's Checklist


PR process:


Studio-specifc:



Testing:

___

### Reviewer's Checklist
#### This section is for reviewers to fill out.
